jobbergate-cli/pyproject.toml

```diff
@@ -8,7 +8,7 @@ packages = [ { include = "jobbergate_cli" } ]
 
 
 [tool.poetry.dependencies]
-python = "^3.6.2"
+python = "^3.8"
```

Comment:
Will this work on CentOS7 with python 3.6?

Reply:
I guess this change is for suiting the unit tests

Reply:
We need to adjust the jobbergate-cli charm to install python3.8.

Python 3.6 is no longer supported.

Reply:
CentOS7 will continue with oython3.6 for another 2 years. We will continue on CentOS7 with python3.6 until them.

Reply:
Here are some additional reasons for using python3.8 beyond official support closing:

1. We are using python3.8 for both license-manager-agent and cluster-agent already. It would be nice to have a consistent baseline version of python for our projects. We already have the mechanics set up for installing python3.8 in our charms.
2. Python 3.8 added some features that I used in this change:
  - Positional-only parameters (using the `*` arg in a function signature)
  - Self-documenting expressions (the `f"{id=}"` syntax)
  - importlib.metadata for retrieving package version of jobbergate-cli (--version flag added in latest commit)
3. Some packages don't support Python 3.6 any more
  - pytest-mock
  - py-buzz 3.0 and snick 1.2 (these are my packages - I wanted to use some newer features from py-buzz 3.0)

Reply:
> Here are some additional reasons for using python3.8 beyond official support closing:

python3.6 is still officially supported on CentOS7. It was not deprecated. It is not going away. (python2.7 is also officially supported, but that is totally another story...)

> We are using python3.8 for both license-manager-agent and cluster-agent already. It would be nice to have a consistent baseline version of python for our projects.

The charm code is python3.6 and this is not going away. If we want a "consistent baseline version", it has to be 3.6. The slurm-charms are a nice example of backwards compatible code.

> We already have the mechanics set up for installing python3.8 in our charms.

1. "We _can_ do xxxx" does not imply "we _should_ do xxxx¨
2. I was against doing that and I am still against it. We are totally running an unsupported python version for those projects, we are not checking for any security issues, and we do not have a QA system to actually validate it. This is dangerous waters.


We do have a strict consideration for resource usage for jobbergate (CPU, memory, disk usage). In order for us considering installing it on other partitions than the `login`, the system *must* be lightweight. Bundling a whole new python distribution goes completely against this.

It appears that python3.8 is only necessary for the tests, is there a problem to run the tests on python3.8 and still make the code compatible with 3.6?

I am not a big fan of using python3.6, but it is what it is...

Reply:
My take on this:

@heitorPB's opinion is completely understandable and makes a lot of sense, but (there is always a but) I don't see any reason of backing up this PR and letting license manager agent and the cluster agent using python 3.8. What's the sense? Did our customers complain with this usage? Do they think there is this huge security problem on their systems? That's the questions we need answers to have a good clearence on this subject.

If the customers clearly say "hey, we do not want this new package (python 3.8) in out systems" then I'm totally against moving with this PR, otherwise, I don't see standardization on having some code with 3.8 and other with 3.6

Reply:
Reverted to python3.6 in latest commit.